capstone-project/src/main/angular-webapp/src/app/map/map.component.spec.ts

```diff
@@ -22,4 +47,73 @@ describe('MapComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should center on user location', () => {
+    spyOn(navigator.geolocation, 'getCurrentPosition').and.callFake(function(success) {
+      const position = {
+        coords: {
+          accuracy: 1,
+          altitude: 1,
+          altitudeAccuracy: 1,
+          heading: 1,
+          speed: 0,
+          latitude: 20,
+          longitude: 85
+        }, timestamp: 1
+      };
+      success(position);
+    });
+
+    component.focusOnUserLocation();
+    const location = new google.maps.LatLng(20, 85)
+    
+    expect(component["gMap"].getCenter().lat()).toEqual(location.lat());
+    expect(component["gMap"].getCenter().lng()).toEqual(location.lng());
+  });
+
+  it('should handle location error', () => {
+    spyOn(navigator.geolocation, 'getCurrentPosition').and.callFake(function (success, error) {
+      const err = {
+        code: 0,
+        PERMISSION_DENIED: 1,
+        POSITION_UNAVAILABLE: 2,
+        TIMEOUT: 3,
+        message: ""
+      };
+      error(err);
+    });
+
+    component.focusOnUserLocation();
+
+    expect(component["gMap"].getCenter()).toEqual(MapComponent["defaultMapCenter"]);
```

Comment:
Overall nice test, 

I don't like this way to access a private field, I'd rather you make 'defaultMapCenter' protected